code/go/0chain.net/smartcontract/zcnsc/stakepool.go

```diff
@@ -104,6 +105,41 @@ func (zcn *ZCNSmartContract) getStakePool(authorizerID datastore.Key, balances c
 	return sp, nil
 }
 
+func (zcn *ZCNSmartContract) GetStakePool(providerType spenum.Provider, providerID datastore.Key, balances chainstate.CommonStateContextI) (sp *StakePool, err error) {
```

Comment:
We should keep methods private if they are not used by other packages. Actually we don't need this method as we do the things in the `getStakePoolAdapter`.